%YAML:1.0

# Project
project_name: "lvi_sam"

#common parameters  need change 
imu_topic: "/imu_raw"
image_topic: "/camera/image_raw"
point_cloud_topic: "lvi_sam/lidar/deskew/cloud_deskewed"

# Lidar Params
use_lidar: 1                     # whether use depth info from lidar or not
lidar_skip: 3                    # skip this amount of scans
align_camera_lidar_estimation: 1 # align camera and lidar estimation for visualization

# lidar to camera extrinsic  need change   lidar frame z towards  -1.5707963
# lidar_to_cam_tx: 0.0
# lidar_to_cam_ty: -0.1377
# lidar_to_cam_tz: -0.3
# # lidar_to_cam_rx: 0.0  #eventhought is not accuracy, it just is fine
# # lidar_to_cam_ry: 0.0  #eventhought is not accuracy, it just is fine
# lidar_to_cam_rz: 0.0

# lidar_to_cam_rx: 1.5707963
# lidar_to_cam_ry: -1.5707963

lidar_to_imu_tx: 0.0
lidar_to_imu_ty: 0.0
lidar_to_imu_tz: 0.6377
lidar_to_imu_rx: 0.0
lidar_to_imu_ry: 0.0
lidar_to_imu_rz: 0.0

# camera model   need change
model_type: PINHOLE
camera_name: camera

# Mono camera config   need change
image_width: 640
image_height: 480
distortion_parameters:
   k1: 0
   k2: 0
   p1: 0
   p2: 0
projection_parameters:
   fx: 320.0020245807715
   fy: 320.0020245807715
   cx: 320.5
   cy: 240.5
fisheye_mask: "/config/fisheye_mask_720x540.jpg"

#imu parameters       The more accurate parameters you provide, the worse performance
acc_n: 0.1       # accelerometer measurement noise standard deviation. #0.2   0.04
gyr_n: 0.01        # gyroscope measurement noise standard deviation.     #0.05  0.004
acc_w: 0.001            # accelerometer bias random work noise standard deviation.  #0.02    can not be zero
gyr_w: 0.0001      # gyroscope bias random work noise standard deviation.     #4.0e-5   can not be zero
g_norm: 9.80     # gravity magnitude

# Extrinsic parameter between IMU and Camera.
estimate_extrinsic: 0   # 0  Have an accurate extrinsic parameters. We will trust the following imu^R_cam, imu^T_cam, don't change it.
                        # 1  Have an initial guess about extrinsic parameters. We will optimize around your initial guess.
                        # 2  Don't know anything about extrinsic parameters. You don't need to give R,T. We will try to calibrate it. Do some rotation movement at beginning. 
#Rotation from camera frame to imu frame, imu^R_cam
extrinsicRotation: !!opencv-matrix
   rows: 3
   cols: 3
   dt: d
   data: [ 0, 0, 1,  
           -1, 0, 0,
           0, -1, 0]
   # data: [ 0, 0, -1, 
   #         -1, 0, 0, 
   #         0, 1, 0]


#Translation from camera frame to imu frame, imu^T_cam
extrinsicTranslation: !!opencv-matrix
   rows: 3
   cols: 1
   dt: d
   data: [0.3, 0.0, 0.5]

#feature traker paprameters
max_cnt: 150            # max feature number in feature tracking
min_dist: 20            # min distance between two features 
freq: 20                # frequence (Hz) of publish tracking result. At least 10Hz for good estimation. If set 0, the frequence will be same as raw image 
F_threshold: 1.0        # ransac threshold (pixel)
show_track: 1           # publish tracking image as topic
equalize: 1             # if image is too dark or light, trun on equalize to find enough features
fisheye: 0              # if using fisheye, trun on it. A circle mask will be loaded to remove edge noisy points

#optimization parameters
max_solver_time: 0.035   # max solver itration time (ms), to guarantee real time
max_num_iterations: 10   # max solver itrations, to guarantee real time
keyframe_parallax: 10.0  # keyframe selection threshold (pixel)

#unsynchronization parameters
estimate_td: 1           # online estimate time offset between camera and imu
td: 0                    # initial value of time offset. unit: s. readed image clock + td = real image clock (IMU clock)

#rolling shutter parameters
rolling_shutter: 0       # 0: global shutter camera, 1: rolling shutter camera
rolling_shutter_tr: 0    # unit: s. rolling shutter read out time per frame (from data sheet). 

#loop closure parameters
loop_closure: 1                    # start loop closure
skip_time: 0.0
skip_dist: 0.0
debug_image: 0                      # save raw image in loop detector for visualization prupose; you can close this function by setting 0 
match_image_scale: 0.5
vocabulary_file: "/config/brief_k10L6.bin"
brief_pattern_file: "/config/brief_pattern.yml"
